File: create_test_pdf.js
// Simple script to create a test PDF using Node.js
const fs = require('fs');
const path = require('path');

// Create a simple HTML content that can be converted to PDF
const htmlContent = `<!DOCTYPE html>
<html>
<head>
    <meta charset="UTF-8">
    <title>Test Bank Statement</title>
    <style>
        body {
            font-family: Arial, sans-serif;
            padding: 40px;
            max-width: 800px;
            margin: 0 auto;
        }
        h1 {
            color: #2c3e50;
            border-bottom: 3px solid #3498db;
            padding-bottom: 10px;
        }
        .header {
            margin-bottom: 30px;
        }
        .account-info {
            background: #ecf0f1;
            padding: 15px;
            border-radius: 5px;
            margin-bottom: 20px;
        }
        table {
            width: 100%;
            border-collapse: collapse;
            margin-top: 20px;
        }
        th, td {
            padding: 12px;
            text-align: left;
            border-bottom: 1px solid #ddd;
        }
        th {
            background-color: #3498db;
            color: white;
        }
        .balance {
            font-size: 18px;
            font-weight: bold;
            color: #27ae60;
            margin-top: 20px;
        }
        .footer {
            margin-top: 30px;
            font-size: 12px;
            color: #7f8c8d;
        }
    </style>
</head>
<body>
    <div class="header">
        <h1>Bank Statement</h1>
        <p><strong>Statement Period:</strong> November 1, 2025 - November 30, 2025</p>
    </div>
    
    <div class="account-info">
        <p><strong>Account Number:</strong> 100002345</p>
        <p><strong>Account Holder:</strong> John Doe</p>
        <p><strong>Account Type:</strong> Checking Account</p>
    </div>
    
    <table>
        <thead>
            <tr>
                <th>Date</th>
                <th>Description</th>
                <th>Debit</th>
                <th>Credit</th>
                <th>Balance</th>
            </tr>
        </thead>
        <tbody>
            <tr>
                <td>2025-11-01</td>
                <td>Opening Balance</td>
                <td>-</td>
                <td>-</td>
                <td>$3,200.00</td>
            </tr>
            <tr>
                <td>2025-11-05</td>
                <td>Transfer to Tom</td>
                <td>$100.00</td>
                <td>-</td>
                <td>$3,100.00</td>
            </tr>
            <tr>
                <td>2025-11-08</td>
                <td>Refund for lunch</td>
                <td>-</td>
                <td>$50.00</td>
                <td>$3,150.00</td>
            </tr>
            <tr>
                <td>2025-11-12</td>
                <td>Refund for voucher</td>
                <td>-</td>
                <td>$20.00</td>
                <td>$3,170.00</td>
            </tr>
            <tr>
                <td>2025-11-15</td>
                <td>May's rent payment</td>
                <td>$750.00</td>
                <td>-</td>
                <td>$2,420.00</td>
            </tr>
            <tr>
                <td>2025-11-20</td>
                <td>Salary deposit</td>
                <td>-</td>
                <td>$2,500.00</td>
                <td>$4,920.00</td>
            </tr>
            <tr>
                <td>2025-11-25</td>
                <td>Grocery store purchase</td>
                <td>$125.50</td>
                <td>-</td>
                <td>$4,794.50</td>
            </tr>
        </tbody>
    </table>
    
    <div class="balance">
        <p>Closing Balance: $4,794.50</p>
    </div>
    
    <div class="footer">
        <p>This is a test document for document extraction testing.</p>
        <p>Generated on: ${new Date().toLocaleDateString()}</p>
    </div>
</body>
</html>`;

// Write HTML file
const htmlPath = path.join(__dirname, 'test_statement.html');
fs.writeFileSync(htmlPath, htmlContent);
console.log('Created test_statement.html');

// Try to convert to PDF using puppeteer if available, otherwise just use the HTML
console.log('\nTest document created!');
console.log('File location:', htmlPath);
console.log('\nYou can:');
console.log('1. Open test_statement.html in a browser and print to PDF');
console.log('2. Or use it directly for testing (HTML files are supported)');
console.log('3. Or install puppeteer to auto-convert: npm install puppeteer');

File: create_test_pdf.py
#!/usr/bin/env python3
"""
Create a test PDF document for extraction testing
"""
try:
    from reportlab.lib.pagesizes import letter
    from reportlab.lib import colors
    from reportlab.lib.units import inch
    from reportlab.platypus import SimpleDocTemplate, Table, TableStyle, Paragraph, Spacer
    from reportlab.lib.styles import getSampleStyleSheet, ParagraphStyle
    from reportlab.lib.enums import TA_CENTER, TA_LEFT
    import os
    
    # Create PDF
    pdf_path = 'test_bank_statement.pdf'
    doc = SimpleDocTemplate(pdf_path, pagesize=letter)
    story = []
    styles = getSampleStyleSheet()
    
    # Title
    title_style = ParagraphStyle(
        'CustomTitle',
        parent=styles['Heading1'],
        fontSize=24,
        textColor=colors.HexColor('#2c3e50'),
        spaceAfter=30,
        alignment=TA_LEFT
    )
    story.append(Paragraph("Bank Statement", title_style))
    story.append(Paragraph("Statement Period: November 1, 2025 - November 30, 2025", styles['Normal']))
    story.append(Spacer(1, 0.3*inch))
    
    # Account Info
    account_data = [
        ['Account Number:', '100002345'],
        ['Account Holder:', 'John Doe'],
        ['Account Type:', 'Checking Account']
    ]
    account_table = Table(account_data, colWidths=[2*inch, 4*inch])
    account_table.setStyle(TableStyle([
        ('BACKGROUND', (0, 0), (0, -1), colors.HexColor('#ecf0f1')),
        ('TEXTCOLOR', (0, 0), (-1, -1), colors.black),
        ('ALIGN', (0, 0), (-1, -1), 'LEFT'),
        ('FONTNAME', (0, 0), (0, -1), 'Helvetica-Bold'),
        ('FONTSIZE', (0, 0), (-1, -1), 10),
        ('BOTTOMPADDING', (0, 0), (-1, -1), 12),
        ('TOPPADDING', (0, 0), (-1, -1), 12),
        ('GRID', (0, 0), (-1, -1), 1, colors.grey)
    ]))
    story.append(account_table)
    story.append(Spacer(1, 0.3*inch))
    
    # Transactions Table
    transactions_data = [
        ['Date', 'Description', 'Debit', 'Credit', 'Balance'],
        ['2025-11-01', 'Opening Balance', '-', '-', '$3,200.00'],
        ['2025-11-05', 'Transfer to Tom', '$100.00', '-', '$3,100.00'],
        ['2025-11-08', 'Refund for lunch', '-', '$50.00', '$3,150.00'],
        ['2025-11-12', 'Refund for voucher', '-', '$20.00', '$3,170.00'],
        ['2025-11-15', "May's rent payment", '$750.00', '-', '$2,420.00'],
        ['2025-11-20', 'Salary deposit', '-', '$2,500.00', '$4,920.00'],
        ['2025-11-25', 'Grocery store purchase', '$125.50', '-', '$4,794.50'],
    ]
    
    transactions_table = Table(transactions_data, colWidths=[1*inch, 2.5*inch, 1*inch, 1*inch, 1*inch])
    transactions_table.setStyle(TableStyle([
        ('BACKGROUND', (0, 0), (-1, 0), colors.HexColor('#3498db')),
        ('TEXTCOLOR', (0, 0), (-1, 0), colors.whitesmoke),
        ('ALIGN', (0, 0), (-1, -1), 'LEFT'),
        ('FONTNAME', (0, 0), (-1, 0), 'Helvetica-Bold'),
        ('FONTSIZE', (0, 0), (-1, 0), 12),
        ('BOTTOMPADDING', (0, 0), (-1, 0), 12),
        ('BACKGROUND', (0, 1), (-1, -1), colors.white),
        ('TEXTCOLOR', (0, 1), (-1, -1), colors.black),
        ('FONTNAME', (0, 1), (-1, -1), 'Helvetica'),
        ('FONTSIZE', (0, 1), (-1, -1), 10),
        ('GRID', (0, 0), (-1, -1), 1, colors.grey),
        ('ROWBACKGROUNDS', (0, 1), (-1, -1), [colors.white, colors.HexColor('#f8f9fa')])
    ]))
    story.append(transactions_table)
    story.append(Spacer(1, 0.3*inch))
    
    # Closing Balance
    balance_style = ParagraphStyle(
        'Balance',
        parent=styles['Normal'],
        fontSize=16,
        textColor=colors.HexColor('#27ae60'),
        spaceBefore=20,
        fontName='Helvetica-Bold'
    )
    story.append(Paragraph("Closing Balance: $4,794.50", balance_style))
    
    # Build PDF
    doc.build(story)
    print(f"✅ Created test PDF: {os.path.abspath(pdf_path)}")
    print(f"📄 File size: {os.path.getsize(pdf_path)} bytes")
    
except ImportError:
    print("⚠️  reportlab not installed. Installing...")
    import subprocess
    import sys
    subprocess.check_call([sys.executable, "-m", "pip", "install", "reportlab", "--quiet"])
    print("✅ Installed reportlab. Please run this script again.")
except Exception as e:
    print(f"❌ Error creating PDF: {e}")
    print("\nAlternative: Use the HTML file and convert manually:")
    print("1. Open test_statement.html in browser")
    print("2. Print → Save as PDF")

